coral-trino/src/main/java/com/linkedin/coral/trino/rel2trino/Calcite2TrinoUDFConverter.java

```diff
@@ -255,6 +262,24 @@ public RexNode visitCall(RexCall call) {
       return adjustReturnTypeWithCast(rexBuilder, super.visitCall(modifiedCall));
     }
 
+    private Optional<RexNode> visitConcat(RexCall call) {
+      // Hive supports operations like CONCAT(date, varchar) while Trino only supports CONCAT(varchar, varchar)
+      // So we need to cast the unsupported types to varchar
```

Comment:
should we also validate that the "unsupported type" is a date? @wmoustafa

Reply:
I don't think that is necessary, the input type has been validated in `StaticHiveFunctionRegistry`, we just need to cast the type to be aligned with Trino's expectation.

Reply:
Thanks for the clarification.